src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -254,7 +254,7 @@ public override BoundNode VisitLambda(BoundLambda node)
             try
             {
                 _factory.CurrentFunction = node.Symbol;
-                var visited = (BoundLambda)base.VisitLambda(node);
+                var visited = (BoundLambda)base.VisitLambda(node)!;
```

Comment:
I would prefer a `Debug.Assert(visited is object);` instead of a suppression.

Reply:
Replaced the suppression with a `Debug.Assert`. An additional warning, but it should be fine. :+1:

Reply:
That additional warning seemed to become an error in testing:
`LocalRewriter.cs(257,31): error CS8600: Converting null literal or possible null value to non-nullable type.`
Reverted it back to what it was for now.

Reply:
It is a warning locally but we push all warnings to errors in the CI runs.

Reply:
The warning is because you need to cast it to `BoundLambda?`, not `BoundLambda`.